instruction.hpp

```diff
@@ -5,6 +5,9 @@
 #include <cstdint>
 #include <iostream>
 #include <functional>
+#include <llvm-16/llvm/IR/Function.h>
+#include <llvm-16/llvm/IR/IRBuilder.h>
+#include <llvm-16/llvm/IR/Value.h>
```

Comment:
Cringe: hardcoded llvm version.
And my system (OpenSuse Leap) has llvm headers in `llvm/...` not `llvm-<version>/llvm/...` (first one seems to be more common)